spec/rubocop/cop/performance/redundant_split_regexp_argument_spec.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Performance::RedundantSplitRegexpArgument do
+  subject(:cop) { described_class.new }
+
+  it 'accepts methods other than split' do
+    expect_no_offenses("'a,b,c'.insert(2, 'a')")
+  end
+
+  it 'accepts when split does not receive a regexp' do
+    expect_no_offenses("'a,b,c'.split(',')")
+  end
+
+  it 'accepts when split does not receive a deterministic regexp' do
+    expect_no_offenses("'a,b,c'.split(/,+/)")
+    expect_no_offenses("'a,b,c'.split(/\./)")
```

Comment:
It seems that the test is passed unexpectedly due to the character escape. The following is used to express the expected `/\./`.

```diff
-    expect_no_offenses("'a,b,c'.split(/\./)")
+    expect_no_offenses('"a,b,c".split(/\./)')
```

or

```diff
-    expect_no_offenses("'a,b,c'.split(/\./)")
+    expect_no_offenses("'a,b,c'.split(/\\./)")
```

Also, `split(/\./)` would be best when an offense will be registered and auto-corrected to `split('.')`.